pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -330,6 +331,9 @@ stdenv.mkDerivation ({
         export NIX_${ghcCommandCaps}="${ghcEnv}/bin/${ghcCommand}"
         export NIX_${ghcCommandCaps}PKG="${ghcEnv}/bin/${ghcCommand}-pkg"
         export NIX_${ghcCommandCaps}_DOCDIR="${ghcEnv}/share/doc/ghc/html"
+        export LD_LIBRARY_PATH="''${LD_LIBRARY_PATH:+''${LD_LIBRARY_PATH}:}${
+          concatStringsSep ":" (map (x: x + "/lib") (filter isDerivation systemBuildInputs))
```

Comment:
1. Nixpkgs has a helper function `makeLibraryPath` that does pretty much what this code does, so it should probably be preferred.

2. Why do you filter on `isDerivation`? `systemBuildInputs` is not supposed to contain anything but derivations, so this seems unnecessary (and misleading to readers of the code).